server/src/main/java/org/elasticsearch/cluster/project/ProjectStateRegistry.java

```diff
@@ -185,26 +220,86 @@ public static Builder builder() {
         return new Builder();
     }
 
+    static class ProjectStatusRegistryDiff implements NamedDiff<Custom> {
```

Comment:
Nit: for consistency with the main class
```suggestion
    static class ProjectStateRegistryDiff implements NamedDiff<Custom> {
```